### ISSUE TITLE
always get mean face during training

### ISSUE THREAD

**Issue by xingmimfl** [microsoft/Deep3DFaceReconstruction#74]
hello, i re-implemented some code, and during the training , landmark loss seems very hard to be smaller once it reaches about 45. and the rendered faces seems to be mean face. Have you ever meet this situation? and what's your method to overcome it ? :P

**Comment by YuDeng** [comment]
You can try to reduce the weights of regularization terms. Or, you can first remove the regularization terms to see how the face is changing during training.

**Comment by xingmimfl** [comment]
> You can try to reduce the weights of regularization terms. Or, you can first remove the regularization terms to see how the face is changing during training.

thx, i'll try it now ~

**Comment by asterisk37n** [comment]
@xingmimfl 
How about it? I have a similar issue. I get almost the same reconstructed face when I put different face image as an input to R-Net.  Though color of faces and positions of eyes and mouths looks good, but especially the direction of the face doesn't get correct. In my case,  it might be because my dataset lacks variations.

**Comment by xingmimfl** [comment]
> @xingmimfl
> How about it? I have a similar issue. I get almost the same reconstructed face when I put different face image as an input to R-Net. Though color of faces and positions of eyes and mouths looks good, but especially the direction of the face doesn't get correct. In my case, it might be because my dataset lacks variations.

see my rep. I solved this problem by using resnet

**Comment by asterisk37n** [comment]
@xingmimfl 
Thank you. I also use ResNet50 as the paper says.

**Comment by ZJH2233** [comment]
> @xingmimfl
> Thank you. I also use ResNet50 as the paper says.
> EDIT: I found I was using frozen ResNet. After I implemented trainable ResNet, it started to train finely.

Hi,I also want to train by myself.But author dosn't provide training code.So, I hope can get your code to reference and study.Thanks!

**Comment by ZJH2233** [comment]
Hi,I also want to train by myself.But author dosn't provide training code.So, I hope can get your code to reference and study.Thanks!

**Comment by asterisk37n** [comment]
@ZJH2233 
See my gist (Paper says that it used a pretrained ResNet-50 on Imagenet, but I couldn't find a pretrained ResNet to train again with tensorflow<=1.12).
https://gist.github.com/asterisk37n/b85ddc49b4d4e9cfb85e92e85c05fab2

**Comment by ZJH2233** [comment]
@asterisk37n
Thank you very much!
The loss function is very complicated for me.Do you write this part code?

About pretrained ResNet ,author Upload a pre-trained model with white light assumption as described in the paper.2020.6
I haven't learned transfer learning so I don't know if that's what you need.

**Comment by asterisk37n** [comment]
@asterisk37n Yes, I wrote loss functions and regularizations.

**Comment by HoiM** [comment]
I also encountered the same problem today. I implemented the training phase with pytorch and pytorch3d. My training result is kind of like an over-fitting. The results on the training set are pretty good but the results on the test set are all mean faces.

I'm not sure if this is really due to the network structure. I used ResNeXt-50. Now I started a new training process with resnet-50. 

By the way, should weight decay be applied to this training process? I found the parameters in the overfit model pretty small and wonder if the overfitting was due to weight decay.
